_sources/vectors_in_cpp/mixed_up_code.rst

```diff
@@ -0,0 +1,292 @@
+Mixed Up Code Exercises
+-----------------------
+
+These questions are **completely OPTIONAL**. We provide these exercises to give you more practice with the material that you've learned. 
+Additionally, they may be a helpful resource when reviewing for assessments. Some lines contain **mistakes** or are **unnecessary** for 
+the function - these lines should not be selected. Make sure to place the blocks at the right indentation levels!
+
+.. parsonsprob:: vectors_in_cpp_mixed_up_code_1
+    :numbered: left
+    :adaptive:
+
+    Construct a block of code that loops over a vector called ``numbers``, initialized as ``{1,2,3,4,5}``,
+    and transforms the vector so each element is doubled. Put the necessary blocks of code in the correct order.
+    -----
+    #include <vector>
+    using namespace std;
+    =====
+    int main() {
+    =====
+        vector<int> numbers = {1, 2, 3, 4, 5};
+    =====
+        vector numbers = {1, 2, 3, 4, 5}; #paired
+    =====
+        for (int i = 0; i < numbers.size(); i++) {
+    =====
+        for (int i = 0; i <= numbers.size(); i++) { #distractor
+    =====
+            numbers[i] *= 2;
+    =====
+        }
+    =====
+    }
+
+.. parsonsprob:: vectors_in_cpp_mixed_up_code_2
+    :numbered: left
+    :adaptive:
+
+    Suppose you have the vector ``vector<string> words = {"car", "cat", "switch", "princess"}``
+    Construct a block of code that transforms the vector to ``vector<string> words = {"cAr", "cAt", "switch", "mArio"}``.
+    Put the necessary blocks of code in the correct order.
+    -----
+    #include <vector>
+    #include <string>
+    using namespace std;
+    =====
+    int main() {
+    =====
+        vector<string> words = {"car", "cat", "switch", "princess"}
+    =====
+        words.pop_back();
+    =====
+        words.push_back("mario");
+    =====
+        for (int i = 0; i < words.size(); i++) {
+    =====
+            for (int j = 0; j < words[i].size(); j++) {
+    =====
+                if (words[i][j] == 'a') {
+    =====
+                    words[i][j] = 'A';
+    =====
+                }
+    =====
+            }
```

Comment:
I mixed up these last three ending brackets and it won't let me get the answer right - I wonder if there's a way to mark which one is which? It's annoying to miss the problem because of these.